File: services/inventory-service/src/main/java/io/minishop/inventory/repository/InventoryReservationRepository.java
package io.minishop.inventory.repository;

import io.minishop.inventory.domain.InventoryReservation;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface InventoryReservationRepository extends JpaRepository<InventoryReservation, Long> {
    Optional<InventoryReservation> findByOrderIdAndProductId(Long orderId, Long productId);
}

File: services/inventory-service/src/main/kotlin/io/minishop/inventory/repository/InventoryReservationRepository.kt
package io.minishop.inventory.repository

import io.minishop.inventory.domain.InventoryReservation
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface InventoryReservationRepository : JpaRepository<InventoryReservation, Long> {
    fun findByOrderIdAndProductId(orderId: Long, productId: Long): Optional<InventoryReservation>
}
